### PR TITLE
Fix autocompletion for enum members in the core classes

### LINKED ISSUE #84509: Autocompletion does not work for the core class enums
### Godot version

5ee983188de97ae027f9b9c1443438063f708a7e

### System information

Windows 11

### Issue description

Currently, the autocompletion for the enum in the core classes does not work.
Here, for example, I want to get access to `Heuristic` enum or `DiagonalMode` but the autocompletion popup isn't showed:

изображение

### Steps to reproduce

^^

### Minimal reproduction project

No need, too easy to reproduce

### PR DESCRIPTION
**Pull request by Chaosus**
My attempt to fix https://github.com/godotengine/godot/issues/84509

изображение

<details><summary>PS: Also fixed the completion of inner classes (where only the enums or subclasses should be returned)</summary>
<p>

изображение

</p>
</details>

### CODE REVIEW

File: modules/gdscript/gdscript_editor.cpp [hunk 1]
```diff
@@ -1071,14 +1075,16 @@ static void _find_identifiers_in_base(const GDScriptCompletionIdentifier &p_base
...
+				_find_identifiers_in_class(base_type.class_type, p_only_functions, p_types_only, base_type.is_meta_type, false, r_result, p_recursion_depth);
 				// This already finds all parent identifiers, so we are done.
 				base_type = GDScriptParser::DataType();
 			} break;
 			case GDScriptParser::DataType::SCRIPT: {
 				Ref<Script> scr = base_type.script_type;
 				if (scr.is_valid()) {
-					if (!p_only_functions) {
+					if (p_types_only) {
+						// TODO: Need to implement Script::get_script_enum_list and retrieve the enum list from a script.
```
Comment (Calinou) [inline modules/gdscript/gdscript_editor.cpp:1086]: Is this needed to support custom enums from named classes?

**Review by Calinou — APPROVED** [review]
Tested locally, it works as expected. Needs a rebase before it can be merged.

https://github.com/godotengine/godot/assets/180032/b8f65859-2545-4bf4-8205-722ae73e6222

File: modules/gdscript/gdscript_editor.cpp (hunk 1 above)
Comment (Chaosus) [reply modules/gdscript/gdscript_editor.cpp:1086]: Yeah

**Comment by Chaosus** [comment]
Rebased*.

**Review by AThousandShips — APPROVED** [review]

**Pull request merged**